packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -140,13 +140,67 @@ library LibProposing {
         emit BlockProposed(state.nextBlockId++, meta);
     }
 
-    function getProposedBlock(
+    function revertToBlock(
         TaikoData.State storage state,
-        uint256 maxNumBlocks,
-        uint256 id
-    ) internal view returns (TaikoData.ProposedBlock storage) {
-        require(id > state.latestVerifiedId && id < state.nextBlockId, "L1:id");
-        return state.getProposedBlock(maxNumBlocks, id);
+        TaikoData.Config memory config,
+        AddressResolver /*resolver*/,
+        uint64 latestVerifiedHeight,
+        uint64 latestVerifiedId,
+        bytes[] calldata inputs
+    ) public {
+        require(
+            latestVerifiedHeight > 0 &&
+                latestVerifiedHeight < state.latestVerifiedHeight,
+            "L1:latestVerifiedHeight"
+        );
+        require(
+            latestVerifiedId > 0 && latestVerifiedId < state.latestVerifiedId,
+            "L1:latestVerifiedId"
+        );
+
+        state.latestVerifiedHeight = latestVerifiedHeight;
+        state.latestVerifiedId = latestVerifiedId;
```

Comment:
emm, if we can revert a L2 verified block, not sure whether the L1 bridge can handle this kind of reversions well? like a person has successfully withdrawn his L2 ETHs / ERC-20 to L1 since the corresponding L2 block was verified, then that L2 block got reverted, which make that person “minting” those L2 ETHs / ERC-20 back in L2 again?